cmd/sonobuoy/app/gen.go

```diff
@@ -39,21 +42,40 @@ var GenCommand = &cobra.Command{
 }
 
 func init() {
-	args.AddSonobuoyImageFlag(&genopts.Image, GenCommand)
-	args.AddModeFlag(&genopts.ModeName, GenCommand)
-	args.AddNamespaceFlag(&genopts.Namespace, GenCommand)
-
+	AddGenFlags(&genopts, GenCommand)
 	RootCmd.AddCommand(GenCommand)
 }
 
+// AddGenFlags adds generation flags to a command
+func AddGenFlags(gen *ops.GenConfig, cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVarP(
+		&mode, "e2e-mode", "", string(ops.Conformance),
+		fmt.Sprintf("What mode to run sonobuoy in. [%s]", strings.Join(ops.GetModes(), ", ")),
+	)
+	// TODO: This variable default needs saner image defaults from ops.f(n) or config
+	cmd.PersistentFlags().StringVarP(
```

Comment:
I think it'd be better to either use gen.* everywhere or nowhere. Makes it easier to know where to add new options.